### ISSUE TITLE
code review: MaskedGlobalAveragePool1D

### ISSUE THREAD

**Issue by BerenLuthien** [CyberZHG/keras-bert#34]
**Is your feature request related to a problem? Please describe.**
Masked Global Average Pool1D

**Describe the solution you'd like**

Hi Cyber, 
I wrote this piece of code with comments, could you help take a look ? Thanks. I initially used it and seems working fine.

```
class MaskedGlobalAveragePool1D(keras.layers.Layer): 
  
     def __init__(self, **kwargs): 
         super(MaskedGlobalAveragePool1D, self).__init__(**kwargs) 
         self.supports_masking = True 
  
     def compute_mask(self, inputs, mask=None): 
         return None 
  
     def compute_output_shape(self, input_shape): 
         return input_shape[:-2] + (input_shape[-1],) 
  
     def call(self, inputs, mask=None): 
         if mask is not None: 
             mask = K.cast(mask, K.floatx())  # cast mask to float
             inputs *= K.expand_dims(mask, axis=-1) # zero wherever mask=0.0
         return K.mean(inputs, axis=-2) # average through time
```

**Comment by CyberZHG** [comment]
Actually, `GlobalAveragePooling1D` supports masking... 😹

**Comment by CyberZHG** [comment]
And the implementation you provided is not correct since you add extra zeros to the sum, therefore the result will be less than the expected value in most cases.

**Comment by BerenLuthien** [comment]
Thanks :)
